homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -47,31 +49,65 @@
 )
 
 
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor from a config entry created in the integrations UI."""
+    opendata = hass.data[DOMAIN][config_entry.entry_id]
+
+    start = config_entry.data[CONF_START]
+    destination = config_entry.data[CONF_DESTINATION]
+    name = config_entry.data.get(CONF_NAME, f"{start} {destination}")
+
+    async_add_entities(
+        [SwissPublicTransportSensor(opendata, start, destination, name)],
+        update_before_add=True,
+    )
+
+
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Swiss public transport sensor."""
-
-    name = config.get(CONF_NAME)
-    start = config.get(CONF_START)
-    destination = config.get(CONF_DESTINATION)
-
-    session = async_get_clientsession(hass)
-    opendata = OpendataTransport(start, destination, session)
-
-    try:
-        await opendata.async_get_data()
-    except OpendataTransportError:
-        _LOGGER.error(
-            "Check at http://transport.opendata.ch/examples/stationboard.html "
-            "if your station names are valid"
+    """Set up the sensor platform."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=config,
+    )
+    if (
+        result["type"] == FlowResultType.CREATE_ENTRY
+        or result["reason"] == "already_configured"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.7.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Swiss public transport",
+            },
+        )
+    else:
+        async_create_issue(
+            hass,
+            DOMAIN,
+            f"deprecated_yaml_import_issue_${result['reason']}",
+            breaks_in_ha_version="2024.7.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key=f"deprecated_yaml_import_issue_${result['reason']}",
         )
-        return
-
-    async_add_entities([SwissPublicTransportSensor(opendata, start, destination, name)])
 
 
 class SwissPublicTransportSensor(SensorEntity):
```

Comment:
Please add unique id and a device (service) in a follow-up PR

Reply:
I am already working at an extension of this integration and will consider it there. This PR is only to add config_flow functionality.